test/Streamly/Test/Data/Parser.hs

```diff
@@ -199,21 +211,25 @@ takeBetween =
             let p = P.takeBetween m n FL.toList
             r <- run $ try $ S.parse p (S.fromList ls)
             return $ case r of
-                Right xs ->
-                    let parsedLen = Prelude.length xs
-                     in if inputLen >= m && parsedLen >= m && parsedLen <= n
-                        then checkListEqual xs $ Prelude.take parsedLen ls
-                        else property False
+                Right s ->
```

Comment:
cannot use `try` as there will be no exception thrown, so there is nothing to catch. Use runIdentity instead.